ef-ctx/tranor
# Add integration tests

This introduces a fake server that can be replaced by an actual server if the environment variables TSURU_TEST_HOST and TSURU_TEST_TOKEN. The commands in the Makefile ensure that all resources required by the integration tests exist in the tsuru API.

How to test this PR:
1. make sure you have latest version of tsuru-client (`brew install tsuru/tsuru/tsuru`, `apt-get install tsuru-client` or checkout and build)
2. run `tsuru install`
3. run `TSURU_TEST_HOST=http://192.168.99.100:8080 TSURU_TEST_TOKEN=$(tsuru token-show | awk '{print $3}') make integration`

Example:

```
% TSURU_TEST_HOST=http://192.168.99.100:8080 TSURU_TEST_TOKEN=$(tsuru token-show | awk '{print $3}') make integration
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin platform-add python || tsuru-admin platform-update python
Error: Duplicate platform
Step 1 : FROM tsuru/python
Pulling from tsuru/python
Digest: sha256:defc498232ab374d3ff7636c2b291fb37a88e3b2407b054b3b6d6c976d910a11
Status: Image is up to date for tsuru/python:latest
 ---> 9c42a1304b7f
Successfully built 9c42a1304b7f
Platform successfully updated!
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin pool-add -p 'dev\dev.example.com' || true
Pool successfully registered.
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin pool-add -p 'qa\qa.example.com' || true
Pool successfully registered.
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin pool-add -p 'stage\stage.example.com' || true
Pool successfully registered.
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin pool-add -p 'prod\example.com' || true
Pool successfully registered.
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin plan-create -c 128 small || true
Plan successfully created!
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin plan-create -c 256 medium || true
Plan successfully created!
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin plan-create -c 512 huge || true
Plan successfully created!
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru team-create myteam || true
Team "myteam" successfully created!
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru team-create superteam || true
Team "superteam" successfully created!
TSURU_TARGET="${TSURU_TEST_HOST}" TSURU_TOKEN="${TSURU_TEST_TOKEN}" tsuru-admin user-quota-change $(tsuru user-info | grep Email: | awk '{print $2}') -- -1
Quota successfully updated.
go test
PASS
ok      github.com/ef-ctx/tranor    3.976s
```
